File: libs/document-processor/src/repo/core/schema/UpdateDocument.ts
import { z } from 'zod';
import { blueIds } from '../blue-ids';
import { withTypeBlueId } from '@blue-labs/language';
import { JsonPatchEntrySchema } from './JsonPatchEntry';

export const UpdateDocumentSchema = withTypeBlueId(blueIds['Update Document'])(
  z.object({
    changeset: z.array(JsonPatchEntrySchema),
  })
);

export type UpdateDocument = z.infer<typeof UpdateDocumentSchema>;

File: references/java-sdk/src/main/java/blue/language/types/conversation/UpdateDocument.java
package blue.language.types.conversation;

import blue.language.model.Node;
import blue.language.model.TypeBlueId;
import blue.language.types.TypeAlias;

import java.util.List;

@TypeAlias("Conversation/Update Document")
@TypeBlueId("FtHZJzH4hqAoGxFBjsmy1svfT4BwEBB4aHpFSZycZLLa")
public class UpdateDocument {
    public List<Node> changeset;

    public UpdateDocument changeset(List<Node> changeset) {
        this.changeset = changeset;
        return this;
    }
}
